beacon-chain/core/altair/sync_committee.go

```diff
@@ -0,0 +1,163 @@
+package altair
+
+import (
+	"github.com/pkg/errors"
+	types "github.com/prysmaticlabs/eth2-types"
+	"github.com/prysmaticlabs/prysm/beacon-chain/core/helpers"
+	"github.com/prysmaticlabs/prysm/beacon-chain/state"
+	ethpb "github.com/prysmaticlabs/prysm/proto/prysm/v1alpha1"
+	"github.com/prysmaticlabs/prysm/shared/bls"
+	"github.com/prysmaticlabs/prysm/shared/bytesutil"
+	"github.com/prysmaticlabs/prysm/shared/hashutil"
+	"github.com/prysmaticlabs/prysm/shared/mathutil"
+	"github.com/prysmaticlabs/prysm/shared/params"
+)
+
+// ValidateNilSyncContribution validates that the signed contribution
+// is not nil.
+func ValidateNilSyncContribution(s *ethpb.SignedContributionAndProof) error {
+	if s == nil {
+		return errors.New("signed message can't be nil")
+	}
+	if s.Message == nil {
+		return errors.New("signed contribution's message can't be nil")
+	}
+	if s.Message.Contribution == nil {
+		return errors.New("inner contribution can't be nil")
+	}
+	if s.Message.Contribution.AggregationBits == nil {
+		return errors.New("contribution's bitfield can't be nil")
+	}
+	return nil
+}
+
+// NextSyncCommittee returns the next sync committee for a given state.
+//
+// Spec code:
+// def get_next_sync_committee(state: BeaconState) -> SyncCommittee:
+//    """
+//    Return the next sync committee, with possible pubkey duplicates.
+//    """
+//    indices = get_next_sync_committee_indices(state)
+//    pubkeys = [state.validators[index].pubkey for index in indices]
+//    aggregate_pubkey = bls.AggregatePKs(pubkeys)
+//    return SyncCommittee(pubkeys=pubkeys, aggregate_pubkey=aggregate_pubkey)
+func NextSyncCommittee(state state.BeaconStateAltair) (*ethpb.SyncCommittee, error) {
+	indices, err := NextSyncCommitteeIndices(state)
+	if err != nil {
+		return nil, err
+	}
+	pubkeys := make([][]byte, params.BeaconConfig().SyncCommitteeSize)
```

Comment:
Why is len set to params.BeaconConfig().SyncCommitteeSize when this depends on `indices`?

Reply:
`len(indices)` has to be `params.BeaconConfig().SyncCommitteeSize` but using the list length is more idiomatic. I'll change to that